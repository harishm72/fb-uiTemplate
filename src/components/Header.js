import React from "react";

const title = "facebook";

const Header = () => <h1> {title} </h1>;

export default Header;
