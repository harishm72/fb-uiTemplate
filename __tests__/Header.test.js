import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from '../src/components/Header';
import renderer from 'react-test-renderer'

const title = 'facebook'

describe('Header --- Snapshot',()=>{
  it('Rednders the Header Component', () => {
      const renderedValue =  renderer.create(<Header title={title}/>).toJSON()
      expect(renderedValue).toMatchSnapshot();
  });
});

describe('Header --- Shallow Render REACT COMPONENTS',()=>{
  let wrapper

  beforeEach(()=>{
      wrapper = shallow(<Header title={title}/>) 
  })

  it('render the Shallow component', () => {
     expect(wrapper.length).toEqual(1)
  });

  it('has header with tag h1' , () => {
      expect(wrapper.contains(<h1>{title}</h1>)).toBe(true)
  })
  
  it('has header value as facebook' , () => {
    expect(wrapper.find('h1').get(0).props.children).toBe(title)
  })

});