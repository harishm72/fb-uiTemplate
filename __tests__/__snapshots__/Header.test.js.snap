exports[`Header --- Snapshot Rednders the Header Component 1`] = `
<div>
  <h1>
    facebook
  </h1>
</div>
`;
